tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/AsyncMethodInvoker_InvokeBegin_Integration.cs

```diff
@@ -52,15 +52,51 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object
             }
 
             var operationContext = WcfCommon.GetCurrentOperationContext();
+
             if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
             {
-                return new CallTargetState(WcfCommon.CreateScope(operationContextProxy.RequestContext));
+                var requestContext = operationContextProxy.RequestContext;
+
+                // First, capture the active scope
+                var activeScope = Tracer.Instance.InternalActiveScope;
+                var spanContextRaw = DistributedTracer.Instance.GetSpanContextRaw() ?? activeScope?.Span?.Context;
+
+                // Then, create the new scope
+                var scope = WcfCommon.CreateScope(requestContext);
+
+                if (scope != null)
+                {
+                    // Save the scope to retrieve it during the AsyncMethodInvoker.InvokeEnd method
+                    WcfCommon.Scopes.Add(((IDuckType)requestContext).Instance, scope);
+                }
+
+                return new CallTargetState(scope, activeScope, spanContextRaw);
             }
             else
             {
                 return CallTargetState.GetDefault();
             }
         }
+
+        internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+            if (exception != null)
```

Comment:
I had to do it :trollface: 
```suggestion
            if (exception is not null)
```

Reply:
:stare: